File: lv0/polynomial/polynomial.py
def solution(polynomial):
    # 다항식을 각 항으로 분리합니다.
    terms = polynomial.split(' + ')

    # x의 계수와 상수항을 초기화합니다.
    x_coefficient, constant_term = 0, 0

    # 다항식의 각 항을 처리합니다.
    for term in terms:
        # 'x'가 있는 경우, x 항입니다.
        if 'x' in term:
            coefficient = term[:-1]
            # 계수가 없는 경우 1로 간주합니다.
            x_coefficient += int(coefficient) if coefficient else 1
        else:
            # 'x'가 없는 경우, 상수항입니다.
            constant_term = int(term)

    # 결과 문자열을 준비합니다.
    # x의 계수가 1인 경우와 그보다 큰 경우를 처리합니다.
    if x_coefficient > 1:    x_part = f'{x_coefficient}x'
    elif x_coefficient == 1: x_part = 'x'
    else:                    x_part = ''

    # 상수항이 0보다 큰 경우 상수항을 추가합니다.
    if constant_term > 0: constant_part = f' + {constant_term}'
    else:                 constant_part = ''

    # 최종 결과를 반환합니다.
    return x_part + constant_part

File: lv0/polynomial/polynomial.php
<?php
function solution($polynomial) {
    // 다항식을 ' + ' 기준으로 분리합니다.
    $terms = explode(" + ", $polynomial);

    // x 항의 계수와 상수항을 저장할 변수를 초기화합니다.
    $xnum = 0;
    $const = 0;

    // 각 항을 순회하며 x 항과 상수항을 분리하여 계산합니다.
    foreach ($terms as $term) {
        if (is_numeric($term)) { // 항이 숫자인 경우(상수항)
            $const += intval($term);
        } else { // 항이 x를 포함하는 경우
            // 항이 'x'인 경우 계수는 1, 그렇지 않은 경우 숫자 부분을 계수로 사용합니다.
            $xnum += ($term == 'x') ? 1 : intval(substr($term, 0, -1));
        }
    }

    // x 항의 계수와 상수항에 따라 결과 문자열을 반환합니다.
    if ($xnum == 0) {
        return strval($const); // x 항이 없는 경우 상수항만 반환합니다.
    } elseif ($xnum == 1) {
        // x 항의 계수가 1인 경우, 상수항이 0인지 아닌지에 따라 다르게 반환합니다.
        return $const != 0 ? 'x + '.strval($const) : 'x';
    } else {
        // x 항의 계수가 1보다 큰 경우, 상수항이 0인지 아닌지에 따라 다르게 반환합니다.
        return $const != 0 ? strval($xnum).'x + '.strval($const) : strval($xnum).'x';
    }
}

// 테스트
$test = "3x + 7 + x";
echo solution($test); // 결과 출력
